### PR TITLE
MultiSig: Remove duplicate reads

### PR DESCRIPTION
**Pull request by aarshkshah1992**
For #278 .

### CODE REVIEW

File: actors/builtin/multisig/multisig_actor.go [hunk 1]
```diff
@@ -385,23 +402,30 @@ func (a Actor) approveTransaction(rt vmr.Runtime, txnID TxnID, txn *Transaction)
...
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to flush pending transactions")
+
 		return nil
 	})
 
 	return executeTransactionIfApproved(rt, st, txnID, txn)
 }
 
-func (a Actor) getTransaction(rt vmr.Runtime, st State, txnID TxnID, proposalHash []byte, checkHash bool) *Transaction {
+func (a Actor) getTransaction(rt vmr.Runtime, ptx *adt.Map, txnID TxnID, proposalHash []byte, checkHash bool) *Transaction {
```
Comment (anorth) [inline actors/builtin/multisig/multisig_actor.go:423]: Nit: this should not be a method, but a free function.

File: actors/builtin/multisig/multisig_state.go [hunk 2]
```diff
@@ -55,57 +53,15 @@ func (st *State) assertAvailable(currBalance abi.TokenAmount, amountToSpend abi.
...
+func getPendingTransaction(ptx *adt.Map, txnID TxnID) (Transaction, error) {
 	var out Transaction
-	found, err := hm.Get(txnID, &out)
+	found, err := ptx.Get(txnID, &out)
 	if err != nil {
 		return Transaction{}, errors.Wrapf(err, "failed to read transaction")
 	}
 	if !found {
-		return Transaction{}, errors.Errorf("failed to find transaction %v in HAMT %s", txnID, as.PendingTxns)
+		return Transaction{}, errors.Errorf("failed to find transaction %v", txnID)
```
Comment (anorth) [inline actors/builtin/multisig/multisig_state.go:63]: Nit: use `xerrors`.

File: actors/builtin/multisig/multisig_actor.go (hunk 1 above)
Comment (aarshkshah1992) [reply actors/builtin/multisig/multisig_actor.go:423]: This is done.

File: actors/builtin/multisig/multisig_state.go (hunk 2 above)
Comment (aarshkshah1992) [reply actors/builtin/multisig/multisig_state.go:63]: This is done.

**Pull request merged**